spring-boot-project/spring-boot-docs/src/main/kotlin/org/springframework/boot/docs/features/testing/springbootapplications/springmvctests/MyControllerTests.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2012-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.docs.features.testing.springbootapplications.springmvctests
+
+import org.junit.jupiter.api.Test
+import org.mockito.BDDMockito
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers
+
+@WebMvcTest(UserVehicleController::class)
+class MyControllerTests(
+	@Autowired val mvc: MockMvc,
+	@MockBean val userVehicleService: UserVehicleService) {
```

Comment:
AFAIK, there is no ParameterResolver for parameters annotated with MockBean (which, BTW annotates the backing field, and not the parameter). So, unless I missed something, this will fail at runtime. Instead, `userVehicleService` should be a `lateinit var`.


Sorry for just leaving this comment: I stumbled on this almost by accident and felt the need to signal the issue, but I applaud your huge effort to translate all those examples in Kotlin. 👏

Reply:
Thanks, fixed this case in three examples.

Reply:
Oh sorry, I wasn't clear enough. The MockBean annotation is needed, but it needs to annotate the lateinit var:

```
@MockBean
lateinit var userVehicleService: UserVehicleService
```

Reply:
Got it, fixed.